src/daft-functions/src/image/encode.rs

```diff
@@ -0,0 +1,80 @@
+use daft_core::{
+    datatypes::{DataType, Field, ImageFormat},
+    schema::Schema,
+    series::Series,
+};
+
+use common_error::{DaftError, DaftResult};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+/// Container for the keyword arguments for `image_encode`
+/// ex:
+/// ```text
+/// image_encode(input, image_format='png')
+/// ```
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ImageEncode {
+    pub image_format: ImageFormat,
+}
+
+#[typetag::serde]
+impl ScalarUDF for ImageEncode {
```

Comment:
Not relevant to this PR but why do we call this `ScalarUDF` when it is not user defined? Might be a good item to change when we clean up our expressions during clean up week.

Reply:
the idea was that they are "user defined", but these initial ones are managed by us. Eventually exposing an interface for third party udfs using the same underlying system.